devops/jobs/ConfigurationWatcher.groovy

```diff
@@ -0,0 +1,149 @@
+        
+        For example,
+        APPS_TO_CONFIG:
+            ecommerce:
+                app_repo: 'ecommerce'
+                
+        accomplishes the same as the dictionary above.
+    CONFIGURATION_REPO_URL: URL of the configuration GitHub repository (REQUIRED).
+    FOLDER_NAME: the name of the folder in which the downstream per-app jobs will live; defaults to DockerCI.
+    ACCESS_CONTROL: list of github users or groups who will have access to the jobs.
+    
+    Expected credentials - these will normally be set up on the Folder.
+    docker-config-json: a file credential that gives each job in the folder permission to push to DockerHub.
+*/
+
+package devops.jobs
+
+import static org.edx.jenkins.dsl.YAMLHelpers.parseYaml
+import org.yaml.snakeyaml.error.YAMLException
+
+import static org.edx.jenkins.dsl.DevopsConstants.common_read_permissions
+import static org.edx.jenkins.dsl.Constants.common_logrotator
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+
+class ConfigurationWatcher {
+    public static def job = { dslFactory, extraVars ->
+        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/configuration-watcher") {
+
+            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+            assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+
+            def config_branch = 'master'
+
+            // to space separated list as string
+            def apps = extraVars.get("APPS_TO_CONFIG").keySet().join(" ")
+
+            // inject APPS as an environment variable to be used by trigger-builds.sh
+            environmentVariables {
+                env('APPS', apps)
+            }
+
+            wrappers common_wrappers
+
+            logRotator common_logrotator
+
+            def access_control = extraVars.get('ACCESS_CONTROL',[])
+            access_control.each { acl ->
+                common_read_permissions.each { perm ->
+                    authorization {
+                        permission(perm,acl)
+                    }
+                }
+            }
+
+            // for GitHub Webhooks; note that you need to use an https URL and include
+            // the trailing slash
+            properties {
+                githubProjectUrl(extraVars.get("CONFIGURATION_REPO_URL"))
+            }
+
+            scm {
+                // check out edx/configuration repository from GitHub
+                git {
+                    remote {
+                        url(extraVars.get("CONFIGURATION_REPO_URL"))
+                        branch(config_branch)
+                    }
+                    extensions {
+                        relativeTargetDirectory('configuration')
+                    }
+                }
+                
+            }
+
+            triggers {
```

Comment:
I can add that to [here](https://github.com/edx/jenkins-job-dsl-internal/blob/master/src/main/groovy/org/edx/jenkins/dsl/DevopsTasks.groovy), if that seems like the best place. I'm assuming that if I add the triggers { ... } context to one of the libraries, the config_branch variable would need to be defined in the job using the library, right? I figure I should probably change the name to be something a bit more generalizable than config_branch. I'm not positive how most of the existing jobs are structured, but do they generally pre-define the branch they check out of a git repository? I just want to make sure it's not confusing to use because of the variable I've got there.